qiskit_machine_learning/algorithms/classifiers/neural_network_classifier.py

```diff
@@ -119,4 +125,25 @@ def predict(self, X: np.ndarray) -> np.ndarray:  # pylint: disable=invalid-name
     def score(
         self, X: np.ndarray, y: np.ndarray, sample_weight: Optional[np.ndarray] = None
     ) -> float:
+        y = self._convert_categorical(y)
```

Comment:
I guess it is not a good idea to call the same method in `score` as in `fit`:

- In `model.fit()` there should be a call `encoder.fit_transform()` (or `fit` and then `transform`)
- While in `score` only `encoder.transform()` must be applied.

Mixing up these two things in one method may lead to a less readable code.
Surely, `encoder.transform()` must be called in `predict` as well.

Reply:
That would definitely work too. That way we could create the `_target_encoder` attribute in the constructor like your first comment suggests and we would re-fit the encoder everytime we call `model.fit()` as you also noted. The only chance for error would be if the user tries to score their model before training it but I don't see that being very big deal.

There should be no calls to the encoder in `predict` becuase there is no instance of target data.

Reply:
Shame on me, you're right, there are no labels in `predict`. Unless we add support of categorical features.
If a user calls `score` or `predict` before `fit`, there's a check `if self._fit_result is None` that raises an exception.